projects/novo-elements/src/elements/data-table/cell-headers/data-table-checkbox-header-cell.component.ts

```diff
@@ -46,6 +48,8 @@ export class NovoDataTableCheckboxHeaderCell<T> extends CdkHeaderCell implements
       if (event.isPageSizeChange) {
         this.checked = false;
         this.dataTable.selectRows(false);
+        this.dataTable.state.checkRetainment('pageSize');
+        this.dataTable.state.reset(false, true);
```

Comment:
it might be helpful to add/update data-table examples with an example of using this new datatable feature